components/camel-netty4/src/main/java/org/apache/camel/component/netty4/SingleUDPNettyServerBootstrapFactory.java

```diff
@@ -132,27 +135,38 @@ protected void startServerBootstrap() throws Exception {
         bootstrap.option(ChannelOption.SO_BROADCAST, configuration.isBroadcast());
         bootstrap.option(ChannelOption.CONNECT_TIMEOUT_MILLIS, configuration.getConnectTimeout());
         
-        // TODO need to find the right setting of below option
         // only set this if user has specified
-        /*
         if (configuration.getReceiveBufferSizePredictor() > 0) {
-            bootstrap.setOption("receiveBufferSizePredictorFactory",
-                    new FixedReceiveBufferSizePredictorFactory(configuration.getReceiveBufferSizePredictor()));
-        }*/
+            bootstrap.option(ChannelOption.RCVBUF_ALLOCATOR,
+                    new FixedRecvByteBufAllocator(configuration.getReceiveBufferSizePredictor()));
+        }
         
         if (configuration.getBacklog() > 0) {
             bootstrap.option(ChannelOption.SO_BACKLOG, configuration.getBacklog());
         }
 
-        //TODO need to check the additional netty options
-        /*
-        if (configuration.getOptions() != null) {
-            for (Map.Entry<String, Object> entry : configuration.getOptions().entrySet()) {
-                connectionlessBootstrap.setOption(entry.getKey(), entry.getValue());
+        Map<String, Object> options = configuration.getOptions();
+        if (options != null) {
+            for (Map.Entry<String, Object> entry : options.entrySet()) {
+                Object value = entry.getValue();
+                ChannelOption<Object> option = ChannelOption.valueOf(entry.getKey());
+                //For all netty options that aren't of type String
+                //TODO: find a way to add primitive Netty options without having to add them to the Camel registry.
+                if (EndpointHelper.isReferenceParameter(value.toString())) {
+                    String name =  ((String)value).substring(1);
+                    Object o = camelContext.getRegistry().lookupByName(name);
+                    if (o == null) {
+                        LOG.warn("Error finding service with name [{}] in the Camel registry", name);
```

Comment:
Dont WARN but use mandatoryLookup as if the option is missing then we should fail.